src/coreclr/vm/ceeload.cpp

```diff
@@ -11370,6 +11377,9 @@ void SaveManagedCommandLine(LPCWSTR pwzAssemblyPath, int argc, LPCWSTR *argv)
     // Get the command line.
     LPCWSTR osCommandLine = GetCommandLineW();
 
+    // store the entrypoint path
+    s_pEntrypointAssemblyPath = pwzAssemblyPath;
```

Comment:
This may not be sufficient for certain scenarios, specifically: superhost and self-extracting host deployments. @agocke do you have any ideas off the top of your head (or know someone who might know) where I can reliably get the path of the entrypoint assembly? I understand this value may be null/empty if the entrypoiny assembly is an in-memory assembly (such as the host scenarios above).

Reply:
Maybe it would be worthwhile to add another field that is the assembly name (the name of the assembly from its Assembly table in its metadata), which I'd image would be known even in the superhost or self-extracting scenarios. However, I don't know if you have easy access to that information from the perspective of the command server.

Reply:
@jander-msft do you think we need both, or could we do one or the other?

Reply:
`GetAppDomain()->m_pRootAssembly->GetFile()->GetPath`?

Reply:
Thanks, @agocke! What are the semantics of that for superhost or bundled hosts? Will the value of the `Path` member be valid?  I'll go take a look 😄

Reply:
@josalem, I spoke with the dotnet-monitor folks and we think that the assembly name (from the assembly identity) is most important, which should be available in all situations where there is an entrypoint assembly. If you are able to get that, and it's not too much trouble, maybe getting the remainder of the identity information would be nice for the ability to look it up in other uses (e.g. a full module list and we need to differentiate it from other modules with the same name).

Reply:
I'll switch to just the name, since that will be more consistent across hosting scenarios.

> a full module list and we need to differentiate it from other modules with the same name

If the goal is a user being able to identify a specific process _a priori_, I'm not sure this information will be something a user would provide or be able to provide. If a user can't specify a process from the entrypoint assembly name, I'm not sure we should be attempting to do any further identification.

Reply:
This should return an empty string if the assembly doesn't have a path.

Reply:
I switched to `m_image_get_assembly_name (mono_assembly_get_main ()->image)` on Mono and `GetAppDomain ()->GetRootAssembly ()->GetSimpleName ()` on CoreCLR.  These should consistently available across hosting scenarios.

Reply:
> I'll switch to just the name, since that will be more consistent across hosting scenarios.
> 
> > a full module list and we need to differentiate it from other modules with the same name
> 
> If the goal is a user being able to identify a specific process _a priori_, I'm not sure this information will be something a user would provide or be able to provide. If a user can't specify a process from the entrypoint assembly name, I'm not sure we should be attempting to do any further identification.

I think I didn't explain my though correctly. We would only allow identification by the assembly name and not be able to differentiate beyond that (in respect to the assembly identity). 

I was saying that it might be useful for the ProcessInfo2 command to return the other assembly identity information for future purposes, if it is cheap to get now (I assume it is cheap to get the other identity information if you are getting the name from the identity). If it is not cheap or if you feel its not worth to do now, then that's fine. We might ask for it in the future, but I don't have a concrete scenario where it might be useful at this moment. It just might be useful to know the entire precise identity of the entry point in general.